src/Renderer/ThreeExtended/PntsLoader.js

```diff
@@ -0,0 +1,98 @@
+import * as THREE from 'three';
+
+const textDecoder = new TextDecoder('utf-8');
+export default {
+    parse: function parse(buffer) {
+        if (!buffer) {
+            throw new Error('No array buffer provided.');
+        }
+        const array = new Uint8Array(buffer);
+        const view = new DataView(buffer);
+
+        let byteOffset = 0;
+        const pntsHeader = {};
+
+        // Magic type is unsigned char [4]
+        pntsHeader.magic = textDecoder.decode(new Uint8Array(array.subarray(byteOffset, 4)));
+        byteOffset += 4;
+
+        if (pntsHeader.magic) {
+            // Version, byteLength, batchTableJSONByteLength, batchTableBinaryByteLength and batchTable types are uint32
+            pntsHeader.version = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.byteLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.FTJSONLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.FTBinaryLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.BTJSONLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            pntsHeader.BTBinaryLength = view.getUint32(byteOffset, true);
+            byteOffset += Uint32Array.BYTES_PER_ELEMENT;
+
+            // binary table
+            if (pntsHeader.FTBinaryLength > 0) {
+                return parseFeatureBinary(array, byteOffset, pntsHeader.FTJSONLength, buffer);
+            }
+
+            // batch table
+            if (pntsHeader.BTBinaryLength > 0) {
+                throw new Error('For pnts loader, BTBinaryLength: not yet managed');
+            }
+        } else {
+            throw new Error('Invalid pnts file.');
+        }
+    },
+};
+
+function parseFeatureBinary(array, byteOffset, FTJSONLength, view) {
```

Comment:
`view` and `array` are unneeded here, you can just pass the original buffer.

```js
const subArrayJson = textDecoder.decode(new Uint8Array(array.subarray(byteOffset, FTJSONLength + byteOffset))); 
```
becomes:
```js
const subArrayJson = textDecoder.decode(new Uint8Array(array, byteOffset, FTJSONLength));
```
and you can directly replace `view` with `buffer` below.

IMHO this improves lisibility.